### PR TITLE
Adds yamllinting to Makefile.common

### PR DESCRIPTION
**Pull request by mwasilew2**
* adds yamllint target to Makefile.common for use by all Prometheus projects

Signed-off-by: Michal Wasilewski <mwasilewski@gmx.com>

closes: https://github.com/prometheus/prometheus/issues/8801

### CODE REVIEW

File: Makefile.common [hunk 1]
```diff
@@ -198,6 +198,11 @@ else
 endif
 endif
 
+.PHONY: common-yamllint
+common-yamllint:
+	@echo ">> running yamllint on all YAML files in the repository"
+	docker run --rm -v "${PWD}":/yaml sdesbure/yamllint find ./ -type f \( -iname \*.yml -o -iname \*.yaml \) | xargs yamllint --no-warnings
```
Comment (SuperQ) [inline Makefile.common:204]: We typically prefer to use local commands, as this won't won't work in our CI (CicleCI Docker in Docker won't volume map).

Comment (fhemberger) [reply Makefile.common:204]: As local command, this works fine without find/xargs.
```bash
yamllint .
```

If you just want to lint YAML files added to git:
```bash
git ls-files '*.yml' '*.yaml' | while read -r file ; do yamllint "$$file"; done
```

**Comment by fhemberger** [comment]
Please also add a `.yamllint` config, these settings usually work fine for me:

```yaml
---
extends: default

rules:
  braces:
    max-spaces-inside: 1
    level: error
  brackets:
    max-spaces-inside: 1
    level: error
  commas: disable
  comments: disable
  comments-indentation: disable
  indentation:
    spaces: consistent
  line-length: disable
```

If you don't care if it's
```yaml
foo:
- bar
- baz
```

or

```yaml
foo:
  - bar
  - baz
```

you can also set `indentation: disable`.

Reference: https://yamllint.readthedocs.io/en/stable/rules.html

**Comment by fhemberger** [comment]
This PR has been idle for a while … if everyone is okay with it, I'd like to continue the work and get it merged.

**Comment by mwasilew2** [comment]
Hey @fhemberger , apologies for not getting back to this earlier. I rebased the fork onto main and used a local command with the settings from your comment (which look reasonable). Let me know if this is better and I'll push another commit with fixes for all yamllint errors.

Also, CI hasn't finished yet so I'm not sure if the yamllint command is available in the build env, but let's just wait for it to finish

**Comment by fhemberger** [comment]
Hi @mwasilew2, thanks for updating the PR!

CI builds are failing because of #8928, updating your branch again should solve this issue. `yamllint` is not installed by default in the `circleci/golang:1.16-node` testing image, so try to add it [before this line](https://github.com/prometheus/prometheus/blob/main/.circleci/config.yml#L31) and [this one](https://github.com/prometheus/prometheus/blob/main/.circleci/config.yml#L102):

```yaml
    - run:
        command: sudo apt-get install yamllint
```

(Haven't used CircleCI for a while so you might need to fiddle with it a bit.)

**Comment by mwasilew2** [comment]
The build was failing on golangci-lint timeout, I increased it from 2m to 4m

**Comment by fhemberger** [comment]
Wow, great work so far, thank you!

**Comment by mwasilew2** [comment]
the CI job failed on a tsdb test, but I'm unable to reproduce this locally. @fhemberger could you please take a look and potentially retry the job (I don't have permissions to do that in Circle CI)?

There's no more changes on my side, I think the PR is ready for a review as soon as the CI pipeline passes

**Comment by fhemberger** [comment]
Changes look good to me, thanks again!

@SuperQ PTAL, the failing tsdb test seems unrelated.

**Comment by roidelapluie** [comment]
I think we could stay with:

```
foo:
- bar
```

to lower the diff. WDYT?

I will also ping @SuperQ

**Comment by fhemberger** [comment]
@roidelapluie Well, it's done already. This would mean changing it all back by hand. 🙈
I think it's good to merge as it is: Only a cosmetic change for now and it doesn't impact functionality.

If you [disable whitespace changes](https://github.com/prometheus/prometheus/pull/8802/files?w=1) in the files tab, it's also easier to review.

**Comment by roidelapluie** [comment]
Why by hand? Or we can enforce the old behaviour and run the linter again

**Comment by fhemberger** [comment]
`yamllint` only lints and complains, you have to fix the errors/warnings yourself. 😉 
([yamlfmt](https://pypi.org/project/yamlfmt/) could help here, but it doesn't adhere to a given `yammlint? config.)

**Comment by SuperQ** [comment]
@roidelapluie While I would prefer to have indentation with `indent-sequences: false`, I want to maintain compatibility with `yq`, which doesn't have this formatting option (due to Go yaml limitations IIRC).

So for now, we need to leave `indent-sequences: true` as the default.

And yes, the `yamllint` Python tool doesn't have a "reformat" option, it is lint-only. :-(

Given the popularity of Yaml, the tooling is meh.

**Comment by mwasilew2** [comment]
just for reference, as it might turn out useful to someone in the future, I had to look around a bit for a yaml formatter and ended up using this vscode plugin: https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml , it has a "Format Document" command

**Comment by fhemberger** [comment]
@mwasilew2 @SuperQ Thank you! 🎉

**Comment by juliusv** [comment]
Hi,

So now this Makefile.common got automatically synced to our other repos by prombot, and I got many emails about tests now failing in those repos:

```
>> running yamllint on all YAML files in the repository
yamllint .
make: yamllint: Command not found
make: *** [Makefile.common:204: common-yamllint] Error 127
```

It seems like all other other prerequisites (CircleCI config changes, yamllint config, etc.) were only added to the Prometheus repo, and those are not synched across repos.

**Comment by SuperQ** [comment]
@juliusv Fixed with https://github.com/prometheus/prometheus/pull/8995

**Pull request merged**